config-plugins/withAndroidWhatsappPackage.js

```diff
@@ -0,0 +1,30 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { withAndroidManifest } = require('@expo/config-plugins');
+
+function addIntentToQuery({ mod, intent }) {
+  const copyMod = { ...mod };
+  if (!copyMod.modResults.manifest.queries) {
+    copyMod.modResults.manifest.queries = [{ intent: [] }];
+  } else if (!copyMod.modResults.manifest.queries[0]) {
+    copyMod.modResults.manifest.queries = [];
+  }
+
+  copyMod.modResults.manifest.queries.push({
+    package: [{ $: { 'android:name': intent.actionName } }],
+    ...(intent.dataScheme && { data: [{ $: { 'android:scheme': intent.dataScheme } }] })
+  });
+  return copyMod;
+}
+
+const withAndroidWhatsappPackage = (config) => {
+  const mod = withAndroidManifest(config, (mod) => {
+    return addIntentToQuery({
+      mod,
+      intent: { actionName: 'com.whatsapp' }
+    });
```

Comment:
this is the only difference, right? couldn't mail and whatsapp happen in one method, that calls `addIntentToQuery` twice?

Reply:
if we add both intents in one file, you can also keep `addIntentToQuery` in that one remaining file and do not need to export import

Reply:
I think I can find a way round that.

<img width="574" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/70acaa2b-0b27-486c-8a9f-ee1a7b3a3ac9">

I will update the code in this way. Yes, it will be a bit complex, but this way we can solve everything in one place.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/001f367f34ae03dade0fa83279b273fd37231cee

Reply:
AndroidManifest after updating the code.
|before|after|
|--|--|
<img width="423" alt="SCR-20230703-ksgb" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/ef39aad8-363e-49ee-b437-a53c257306ee"> | <img width="411" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/14c9e199-3bad-4bd0-acd3-d41cdba6649d">